### PR TITLE
HPCC-30974 Copying super files via fileservices/dfu does not track read/cost stats

### PR DESCRIPTION
**Pull request by shamser**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [x] Test my draft Pull Request.

### CODE REVIEW

File: dali/ft/filecopy.cpp [hunk 1]
```diff
@@ -3384,25 +3385,9 @@ void FileSprayer::spray()
     afterTransfer();
 
     //If got here then we have succeeded
+    //(note: if we don't get here, file access costs will not be updated)
```
Comment (shamser) [inline dali/ft/filecopy.cpp:3388]: todo:https://track.hpccsystems.com/browse/HPCC-31032

File: dali/ft/filecopy.cpp [hunk 2]
```diff
@@ -3771,17 +3787,8 @@ void FileSprayer::updateTargetProperties()
...
-        {
-            IPropertyTree & fileAttr = distributedSource->queryAttributes();
-            cost_type legacyReadCost = getLegacyReadCost(fileAttr, distributedSource);
-            cost_type curReadCost = money2cost_type(calcFileAccessCost(distributedSource, 0, totalNumReads));
-            distributedSource->addAttrValue(getDFUQResultFieldName(DFUQRFreadCost), legacyReadCost+curReadCost);
-            distributedSource->addAttrValue(getDFUQResultFieldName(DFUQRFnumDiskReads), totalNumReads);
-        }
-    }
+
+    progressReport->setFileAccessCost(cost_type2money(totalCost));
```
Comment (shamser) [inline dali/ft/filecopy.cpp:3791]: Improvement (read/write cost separately):https://track.hpccsystems.com/browse/HPCC-31032

**Review by jakesmith — COMMENTED** [review]
@shamser - commenting only so not tagged for review.
Please refresh review status when ready for review

File: dali/ft/filecopy.cpp [hunk 3]
```diff
@@ -3384,25 +3384,9 @@ void FileSprayer::spray()
     afterTransfer();
 
     //If got here then we have succeeded
+    //Note: On failure, costs will not be updated.  Future: would be useful to have a way to update costs on failure.
```
Comment (shamser) [inline dali/ft/filecopy.cpp:3387]: future work: https://track.hpccsystems.com/browse/HPCC-31042

File: dali/ft/filecopy.cpp (hunk 2 above)
Comment (jakesmith) [reply dali/ft/filecopy.cpp]: wrong JIRA ?

**Comment by shamser** [comment]
@jakesmith

**Review by jakesmith — APPROVED** [review]
@shamser - looks good, I can't see any issues.

I have opened a separate to revisit this, so that these stats are updated whilst the spray/copy is in-flight (not just when finished):  https://track.hpccsystems.com/browse/HPCC-30965

**Comment by ghalliday** [comment]
@shamser please rebase to avoid the clash with the merged cost_type change

**Review by jakesmith — APPROVED** [review]
@shamser - looks good

**Comment by shamser** [comment]
@jakesmith The regression looks unrelated to this commit.  It looks like a python related problem. @AttilaVamos Does this regression looks familiar?

**Comment by jakesmith** [comment]
> @jakesmith The regression looks unrelated to this commit. It looks like a python related problem. @AttilaVamos Does this regression looks familiar?

agree.
It may already be fixed elsewhere. @GordonSmith may know?

**Pull request merged**